src/advanced-settings/data/api.test.js

```diff
@@ -0,0 +1,102 @@
+import { getAuthenticatedHttpClient } from '@edx/frontend-platform/auth';
+import {
+  getConfig, camelCaseObject, modifyObjectKeys, snakeCaseObject,
+} from '@edx/frontend-platform';
+import {
+  getCourseAdvancedSettings,
+  updateCourseAdvancedSettings,
+  getProctoringExamErrors,
+} from './api';
+import { convertObjectToSnakeCase } from '../../utils';
+
+jest.mock('@edx/frontend-platform', () => ({
+  getConfig: jest.fn(),
+  camelCaseObject: jest.fn(),
+  modifyObjectKeys: jest.fn(),
+  snakeCaseObject: jest.fn(),
+}));
+
+jest.mock('@edx/frontend-platform/auth', () => ({
+  getAuthenticatedHttpClient: jest.fn(),
+}));
+
+jest.mock('../../utils', () => ({
+  convertObjectToSnakeCase: jest.fn(),
+}));
+
+describe('courseSettings API', () => {
+  const mockHttpClient = {
+    get: jest.fn(),
+    patch: jest.fn(),
+  };
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+    getAuthenticatedHttpClient.mockReturnValue(mockHttpClient);
+    getConfig.mockReturnValue({ STUDIO_BASE_URL: 'http://studio.test' });
+  });
+
+  describe('getCourseAdvancedSettings', () => {
+    it('should fetch and format course advanced settings', async () => {
+      const fakeData = { key: { value: 'some_value' } };
+      const camelCased = { key: { value: 'some_value' } };
+      const modified = { key: { value: 'some_value_snake' } };
+
+      mockHttpClient.get.mockResolvedValue({ data: fakeData });
+      camelCaseObject.mockReturnValue(camelCased);
+      snakeCaseObject.mockReturnValue('some_value_snake');
+      modifyObjectKeys.mockImplementation((obj, fn) => {
+        Object.keys(obj).forEach(fn);
+        return modified;
```

Comment:
This does not look like a proper mock for `modifyObjectKeys`.  Either we actually loop through the inputs and modify each one, or we just forego mocking this function.

Do we actually need to mock these frontend-platform utility functions, though?  I feel like we might not.